File: first_page_module/src/main/java/com/mobile/first_page_module/model/FirstPageModelImpl.java
package com.mobile.first_page_module.model;

import com.mobile.common_library.manager.RetrofitManager;
import com.mobile.first_page_module.request.FirstPageRequest;

import java.util.Map;

import io.reactivex.Observable;
import okhttp3.ResponseBody;

/**
 * author    : Urasaki
 * e-mail    : Urasaki@qq.com
 * date      : 2019/3/10 17:03
 * introduce :
 */

public class FirstPageModelImpl implements IFirstPageModel {
	
	private static final String TAG = "FirstPageModelImpl";
	
	public FirstPageModelImpl() {
	}
	
	@Override
	public Observable<ResponseBody> firstPage(Map<String, String> bodyParams) {
		return RetrofitManager.getInstance().getRetrofit()
				   .create(FirstPageRequest.class)
				   .getFirstPage(bodyParams);
	}
	
	@Override
	public Observable<ResponseBody> firstPageDetails(Map<String, String> bodyParams) {
		return RetrofitManager.getInstance().getRetrofit()
				   .create(FirstPageRequest.class)
				   .getFirstPageDetails(bodyParams);
	}
	
	//    @Override
	//    public Observable<FirstPageResponse.QuestionBean> firstPageData(Map<String, String> bodyParams) {
	//        return RetrofitManager.getInstance().getRetrofit()
	//                .create(FirstPageRequest.class)
	//                .getFirstPageData(bodyParams);
	//    }
	
}

File: first_page_module/src/main/java/com/phone/first_page_module/model/FirstPageModelImpl.kt
package com.phone.first_page_module.model

import com.phone.common_library.manager.RetrofitManager
import com.phone.first_page_module.request.FirstPageRequest
import io.reactivex.Observable
import okhttp3.ResponseBody
import java.util.*

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2019/3/10 17:03
 * introduce :
 */

class FirstPageModelImpl : IFirstPageModel {

    private val TAG = "FirstPageModelImpl"

    override fun firstPage(bodyParams: Map<String?, String?>?): Observable<ResponseBody?> {
        return RetrofitManager.get().getRetrofit()
            .create(FirstPageRequest::class.java)
            .getFirstPage(bodyParams)
    }

    override fun firstPageDetails(bodyParams: Map<String?, String?>?): Observable<ResponseBody?> {
        return RetrofitManager.get().getRetrofit()
            .create(FirstPageRequest::class.java)
            .getFirstPageDetails(bodyParams)
    }

    //    @Override
    //    public Observable<FirstPageResponse.QuestionBean> firstPageData(Map<String, String> bodyParams) {
    //        return RetrofitManager.get().getRetrofit()
    //                .create(FirstPageRequest.class)
    //                .getFirstPageData(bodyParams);
    //    }

}